pdf_craft/pdf/page_extractor.py

```
@@ -197,6 +176,74 @@ def image2page(
+            if has_repetitive_ngrams(
+                text, min_ngram=2, max_ngram=5, repeat_threshold=16
+            ):
+                continue
+
+            # 检测长模式重复（保守策略）
+            if has_repetitive_ngrams(
+                text, min_ngram=6, max_ngram=20, repeat_threshold=8
+            ):
+                continue
+
+            hash: str | None = None
```

Ruff (0.16.1)

[error] Variable `hash` is shadowing a Python builtin

(A001)